packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -502,6 +511,18 @@ export class Function extends FunctionBase {
     const code = props.code.bind(this);
     verifyCodeConfig(code, props.runtime);
 
+    let profilingGroupEnvironmentVariables = {};
+    if (props.profiling) {
+      this.profilingGroup = new ProfilingGroup(this, 'ProfilingGroup');
+      this.profilingGroup.grantPublish(this.role);
+      profilingGroupEnvironmentVariables = {
+        AWS_CODEGURU_PROFILER_GROUP_ARN: this.profilingGroup.profilingGroupArn,
+        AWS_CODEGURU_PROFILER_ENABLED: 'TRUE',
+      };
+    }
+
+    this.environment = { ...(props.environment || {}), ...profilingGroupEnvironmentVariables };
```

Comment:
This will overwrite these environment variables if they were also specified by the customer. 

I think the correct behaviour should be to prefer using customer provided ones.

Also, add a unit test to this effect.

Reply:
do you mean a situation where the customer provides the PROFILER related env variables?

Reply:
added code to throw exception if these variables are set with profiling options enabled.